apps/functions/document-publish/publish-document/index.js

```diff
@@ -1,6 +1,57 @@
+import config from './config';
+import got from 'got';
+import { blobClient } from './blob-client';
+
 /**
  * @type {import('@azure/functions').AzureFunction}
  */
-export const index = async (context, document) => {
-	context.log(`Message ${JSON.stringify(document)}`);
+export const index = async (
+	context,
+	{ documentId, documentURI, documentReference, filename, originalFilename }
+) => {
+	context.log(`Publishing document ID ${documentId} at URI ${documentURI}`);
+
+	if (!documentId || !documentURI || !filename || !originalFilename) {
+		// TODO: Once we sort out documentReference, validate that too
+		throw Error('One or more required properties are missing.');
+	}
+
+	const publishFileName = buildPublishedFileName({
+		documentReference,
+		filename,
+		originalFilename
+	});
+
+	context.log(`Deploying to file name ${publishFileName}`);
+
+	await blobClient.copyFile({
+		sourceContainerName: config.BLOB_SOURCE_CONTAINER,
+		sourceBlobName: documentURI,
+		destinationContainerName: config.BLOB_PUBLISH_CONTAINER,
+		destinationBlobName: publishFileName
+	});
+
+	await got
```

Comment:
@chrisdalykinandcarta What would happen if this request failed for whatever reason? Should we consider wrapping in a try catch and reversing any changes made above or does it not matter?

Reply:
Good catch (pun intended), we don't really have a pattern for this at the moment - but if the request fails, we're probably going to assume it's a transient error and bubbling the exception back up to the function runtime will attempt a retry with exponential back-off.

There will potentially be cases where we hit issues we'll want to step back from, just not sure what these are yet. The function itself has to be idempotent so we probably do want to figure out what happens if a document is already in a certain state.